Move godot-cpp custom bindings information from "Getting Started" to build system information (#11344)
godotengine/godot-docs: tutorials/scripting/cpp/build_system/scons.rst

`godot-cpp-template <https://github.com/godotengine/godot-cpp-template>`__ contains an
`example setup <https://github.com/godotengine/godot-cpp-template/tree/main/.github/workflows>`__
for a GitHub based CI workflow.

> Using custom bindings

Reviewer: I'm not sure how I feel about calling this "custom bindings". To me, a "binding" is like godot-cpp or godot-rust -- the whole library that can interact with Godot. This is more like using a custom API file to build your bindings with

Maybe this should be like:

```suggestion
Using a custom API file
```

... or similar?

---------------------

> Every branch of godot-cpp comes with bindings appropriate for the respective Godot version
> (e.g. the ``4.3`` branch comes with bindings compatible with Godot version ``4.3`` and

Reviewer: ```suggestion
Every branch of godot-cpp comes with an API file (``extension_api.json``) appropriate for the respective Godot version
(e.g. the ``4.3`` branch comes with the API file compatible with Godot version ``4.3`` and
```

later).

However, you may want to use custom bindings, for example:

* If you want to use the latest bindings from Godot ``master``.

> * If you want to use bindings exposed by custom modules.

Reviewer: We should also include something allow the lines of "* If you build Godot yourself with different options than the official builds" because you should do this if you built with options like `scons precision=double` or `scons disable_3d=yes` or `scons deprecated=no` as well

Author: Good point!

> However, you may want to use custom bindings, for example:
> 
> * If you want to use the latest bindings from Godot ``master``.
> * If you want to use bindings exposed by custom modules.

Reviewer: ```suggestion
However, you may want to use a custom ``extension_api.json``, for example:

* If you want to use the latest APIs from Godot ``master``.
* If you want to use APIs exposed by custom modules.
```

> To use custom bindings, you first have to generate them from the appropriate Godot

Reviewer: ```suggestion
To use a custom API file, you first have to generate it from the appropriate Godot
```

executable:

.. code-block:: shell

    godot --dump-extension-api

The resulting ``extension_api.json`` file will be created in the executable's

> directory. To use these custom bindings, you can add ``custom_api_file`` to

Reviewer: ```suggestion
directory. To use this, you can add ``custom_api_file`` to
```

your build command:

.. code-block:: shell

    scons platform=<platform> custom_api_file=<PATH_TO_FILE>

> Alternatively, you can add them as the default bindings to your project by adding

Reviewer: ```suggestion
Alternatively, you can add it as the default API file to your project by adding
```
